tests/test_generate_framework_master_csv.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+"""Tests for GenerateMasterCSV class."""
+from collections import OrderedDict
+import json
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+import mock
+
+from dmscripts.generate_framework_master_csv import GenerateMasterCSV
+
+
+def test_get_fieldnames(mock_data_client):
+    """Test building filenames."""
+    csv_builder = GenerateMasterCSV(client=mock_data_client, target_framework_slug='test_framework_slug')
+    assert mock_data_client.get_framework.called_once_with('test_framework_slug')
+
+    csv_builder.service_status_labels = OrderedDict([('key1', 'test_lot_prefix_1'), ('key2', 'test_lot_prefix_2')])
+    csv_builder.static_fieldnames = ('test_static_fieldname_1', 'test_static_fieldname_2')
+
+    expected_data = tuple([
+        'test_static_fieldname_1',
+        'test_static_fieldname_2',
+        'test_lot_prefix_1_test_lot_slug_1',
+        'test_lot_prefix_2_test_lot_slug_1',
+        'test_lot_prefix_1_test_lot_slug_2',
+        'test_lot_prefix_2_test_lot_slug_2'
+    ])
+    assert csv_builder.get_fieldnames() == expected_data
+
+
+@mock.patch(
+    'dmscripts.generate_framework_master_csv.GenerateMasterCSV.get_fieldnames',
+    return_value=['test_field_1', 'test_field_2'],
+)
+def test_write_csv(fieldname_mock, mock_data_client):
+    """Test writing csv."""
+    csv_builder = GenerateMasterCSV(client=mock_data_client, target_framework_slug='test_framework_slug')
+    assert fieldname_mock.called_once_with('test_framework_slug')
+    csv_builder.output = [
+        {'test_field_1': 'foo', 'test_field_2': 'bar'},
+        {'test_field_1': 'baz', 'test_field_2': 'Ąćĉ'}
```

Comment:
So having thought about this one I think it's probably not worth changing. The `from __future__ import unicode_literals` doesn't play nice with the csv library. I think there's also an argument that instead of forcing unicode literals we should aim to support both given we're attempting to support py2 and py3. In all likelihood the code we write will have to deal with both. What do you think?